### PR TITLE
Add CUDA_STATIC_MATH_LIBRARIES

### PR DESCRIPTION
**Pull request by KyleFromNVIDIA**
Usage of the CUDA math libraries is independent of the CUDA runtime. Make their static/shared status separately controllable.

Contributes to https://github.com/rapidsai/build-planning/issues/35

### CODE REVIEW

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -59,7 +59,8 @@ option(CUDA_ENABLE_KERNELINFO "Enable kernel resource usage info" OFF)
 option(CUDA_ENABLE_LINEINFO
        "Enable the -lineinfo option for nvcc (useful for cuda-memcheck / profiler)" OFF
 )
-option(CUDA_STATIC_RUNTIME "Statically link the CUDA toolkit runtime and libraries" OFF)
+option(CUDA_STATIC_RUNTIME "Statically link the CUDA toolkit runtime" OFF)
+option(CUDA_STATIC_MATH_LIBRARIES "Statically link the CUDA toolkit math libraries" OFF)
```
Comment (bdice) [inline cpp/CMakeLists.txt:63]:
```suggestion
option(CUDA_STATIC_RUNTIME "Statically link the CUDA runtime" OFF)
option(CUDA_STATIC_MATH_LIBRARIES "Statically link the CUDA math libraries" OFF)
```

File: cpp/CMakeLists.txt [hunk 2]
```diff
@@ -105,7 +106,10 @@ message(VERBOSE "RAFT: Enable kernel resource usage info: ${CUDA_ENABLE_KERNELIN
 message(VERBOSE "RAFT: Enable lineinfo in nvcc: ${CUDA_ENABLE_LINEINFO}")
 message(VERBOSE "RAFT: Enable nvtx markers: ${RAFT_NVTX}")
 message(VERBOSE
-        "RAFT: Statically link the CUDA toolkit runtime and libraries: ${CUDA_STATIC_RUNTIME}"
+        "RAFT: Statically link the CUDA toolkit runtime: ${CUDA_STATIC_RUNTIME}"
+)
+message(VERBOSE
+        "RAFT: Statically link the CUDA toolkit math libraries: ${CUDA_STATIC_MATH_LIBRARIES}"
```
Comment (bdice) [inline cpp/CMakeLists.txt:112]:
```suggestion
        "RAFT: Statically link the CUDA runtime: ${CUDA_STATIC_RUNTIME}"
)
message(VERBOSE
        "RAFT: Statically link the CUDA math libraries: ${CUDA_STATIC_MATH_LIBRARIES}"
```

**Review by bdice — APPROVED** [review]
Approving with a couple small suggestions.

**Review by robertmaynard — APPROVED** [review]

**Comment by KyleFromNVIDIA** [comment]
I added setting the variable to the Python builds - requesting re-review.

**Pull request merged**